.github/workflows/qa.yaml

```diff
@@ -0,0 +1,40 @@
+name: Quality Assurance
+
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
+    branches: [ "main" ]
+  workflow_dispatch:
+
+permissions:
+  contents: read
+
+jobs:
+  qa:
+    name: Quality Checks
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Git Checkout
+        uses: actions/checkout@v4
+
+      - name: PHP Setup
+        uses: shivammathur/setup-php@v2
+        with:
+          php-version: 8.2
+
+      - name: Validate composer.json
+        run: composer validate --strict
+
+      - name: Install dependencies
+        uses: ramsey/composer-install@v2
+
+      - name: Check CS-Fixer
+        run: composer cs:check
+
+      - name: Check PHPStan
+        run: composer phpstan
+
+      - name: Execute tests
+        run: composer tests
```

Comment:
Do we need this here? They also run in the other pipeline, right?

Reply:
Depends on the matrix. Currently the matrix contains also the 8.2 + highest entry.

So what do you prefer? qa without Tests, or tests only with legacy versions?

Reply:
I prefer QA without tests and Tests with all versions.